notebooks/official/workbench/spark/spark_sample_notebook.ipynb

```diff
@@ -0,0 +1,1213 @@
+{
```

Comment:
<p>Region only used again in mb command(and for users creating a cluster, was set above). Perhaps delete this re-set sesction or refactor it by moving the REGION set section before creating a cluster so all users will set it once.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/720/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Reply:
<p>I used <code>REGION</code>  variable twice, one for the <code>Dataproc cluster</code>  and one for the <code>Cloud Storage bucket</code> . It is because after a user creates the <code>Dataproc cluster</code> and switch the kernel, every variables that are previously declared lost.</p><p>Resetting a kernel is mandatory as this tutorial must use a Dataproc cluster, unless a user already has it and set the right kernel already.</p><p> </p><p>If we remove this <code>Region</code> and want to use only <code>CLUSTER_REGION</code> , which was declared prior to <code>creating a Dataproc cluster</code> ,  <code>Create a Cloud Storage bucket</code> part also goes to before creating a cluster. But I am afraid that they will lose <code>BUCKET_URI</code> in this case, which is required for writing back to BigQuery.</p>